app/cdap/components/common/InlayDrawerWidget/index.tsx

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { Box, IconButton, Typography } from '@material-ui/core';
+import grey from '@material-ui/core/colors/grey';
+import CloseRoundedIcon from '@material-ui/icons/CloseRounded';
+import Menu, { IActionsOptions } from 'components/common/InlayDrawerWidget/Menu';
+import React, { PropsWithChildren } from 'react';
+import styled from 'styled-components';
+
+interface IRecipeStepWidgetProps extends PropsWithChildren<unknown> {
```

Comment:
don't use unknown and don't extend props with children - heres an example of how to do it https://dev.to/typescripttv/typing-react-props-in-typescript-5hal

Reply:
updated the usage of PropsWithChildren as suggested in the document
used Solution 1: PropsWithChildren
removed extends and updated props type in component props to PropsWithChildren<IRecipeStepWidgetProps>